packages/eslint-plugin/src/rules/no-unsafe-call.ts

```diff
@@ -74,6 +75,52 @@ export default createRule<[], MessageIds>({
             type: isErrorType ? '`error` type' : '`any`',
           },
         });
+        return;
+      }
+
+      if (isBuiltinSymbolLike(services.program, type, 'Function')) {
+        // this also matches subtypes of `Function`, like `interface Foo extends Function {}`.
+        //
+        // For weird TS reasons that I don't understand, these are
+        //
+        // safe to construct if:
+        // - they have at least one call signature _that is not void-returning_,
+        // - OR they have at least one construct signature.
+        //
+        // safe to call (including as template) if:
+        // - they have at least one call signature
+        // - OR they have at least one construct signature.
+
+        const constructSignatures = type.getConstructSignatures();
+        if (constructSignatures.length > 0) {
+          return;
+        }
+
+        const callSignatures = type.getCallSignatures();
+        if (messageId === 'unsafeNew') {
+          const nonVoidReturnCallSignatures = callSignatures.filter(
+            signature =>
+              !tsutils.isIntrinsicVoidType(signature.getReturnType()),
+          );
+
+          if (nonVoidReturnCallSignatures.length > 0) {
```

Comment:
[Style] Nit: no need to loop through all of the signatures, even if there are never that many...

```suggestion
          if (
            callSignatures.some(
              signature =>
                !tsutils.isIntrinsicVoidType(signature.getReturnType()),
            )
          ) {
```

Reply:
Ah, yes. One of issues that is super obvious when reading code, but not when coming up with the rule logic 😆